nilearn/_utils/niimage.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import gc
+import collections
+
+import nibabel
+
+
+def _get_shape(img):
+    # Use the fact that Nifti1Image has a shape attribute that is
+    # faster than loading the data from disk
+    if hasattr(img, 'shape'):
+        shape = img.shape
+    else:
+        shape = img.get_data().shape
+    return shape
+
+
+def _safe_get_data(img):
+    """ Get the data in the image without having a side effect on the
+        Nifti1Image object
+    """
+    if hasattr(img, '_data_cache') and img._data_cache is None:
+        # Copy locally the Nift1Image object to avoid the side effect of data
+        # loading
+        img = copy.deepcopy(img)
+    # typically the line below can double memory usage
+    # that's why we invoke a forced call to the garbage collector
+    gc.collect()
+    return img.get_data()
+
+
+def is_img(obj):
```

Comment:
I think it's because it boils down to loading the data itself so doing the equivalent of `check_niimg`.